src/main/kotlin/com/sourcegraph/cody/api/SourcegraphApiRequestExecutor.kt

```diff
@@ -18,160 +16,122 @@ import java.io.InputStream
 
-  abstract class Base() : SourcegraphApiRequestExecutor() {
-    protected fun <T> RequestBuilder.execute(
-        request: SourcegraphApiRequest<T>,
-        indicator: ProgressIndicator
-    ): T {
-      indicator.checkCanceled()
-      try {
-        LOG.debug("Request: ${request.url} ${request.operationName} : Connecting")
-        return connect {
-          val connection = it.connection as HttpURLConnection
-          if (request is SourcegraphApiRequest.WithBody) {
-            LOG.debug(
-                "Request: ${connection.requestMethod} ${connection.url} with body:\n${request.body} : Connected")
-            request.body?.let { body -> it.write(body) }
-          } else {
-            LOG.debug("Request: ${connection.requestMethod} ${connection.url} : Connected")
-          }
-          checkResponseCode(connection)
-          indicator.checkCanceled()
-          val result = request.extractResult(createResponse(it, indicator))
+        request.extractResult(createResponse(it, indicator)).apply {
           LOG.debug("Request: ${connection.requestMethod} ${connection.url} : Result extracted")
-          result
-        }
-      } catch (e: SourcegraphStatusCodeException) {
-        @Suppress("UNCHECKED_CAST")
-        if (request is SourcegraphApiRequest.Get.Optional<*> &&
-            e.statusCode == HttpURLConnection.HTTP_NOT_FOUND)
-            return null as T
-        else throw e
-      } catch (e: SourcegraphConfusingException) {
-        if (request.operationName != null) {
-          val errorText = "Can't ${request.operationName}"
-          e.setDetails(errorText)
-          LOG.debug(errorText, e)
         }
-        throw e
       }
+    } catch (e: SourcegraphStatusCodeException) {
+      throw e
+    } catch (e: SourcegraphConfusingException) {
+      if (request.operationName != null) {
+        val errorText = "Can't ${request.operationName}"
+        e.setDetails(errorText)
+        LOG.debug(errorText, e)
+      }
+      throw e
     }
+  }
 
-    protected fun createRequestBuilder(request: SourcegraphApiRequest<*>): RequestBuilder {
-      return when (request) {
-            is SourcegraphApiRequest.Get -> HttpRequests.request(request.url)
-            is SourcegraphApiRequest.Post -> HttpRequests.post(request.url, request.bodyMimeType)
-            else -> throw UnsupportedOperationException("${request.javaClass} is not supported")
-          }
-          .userAgent("Cody")
-          .throwStatusCodeException(false)
-          .forceHttps(false)
-          .accept(request.acceptMimeType)
-    }
+  private fun createRequestBuilder(request: SourcegraphApiRequest<*>): RequestBuilder {
+    return when (request) {
+          is SourcegraphApiRequest.Get -> HttpRequests.request(request.url)
+          is SourcegraphApiRequest.Post -> HttpRequests.post(request.url, request.bodyMimeType)
+        }
+        .userAgent("Cody")
+        .throwStatusCodeException(false)
+        .forceHttps(false)
```

Comment:
I see the former version had `.accept(request.acceptMimeType)` but it is not present here. Is it ok?

Reply:
It was null by default and never set anywhere, so this should be fine. The green path worked fine, do you think there might be a default value that null unsets perhaps?

Reply:
Have confirmed this is fine, we never set it on `main` branch, so we're just passing in null when the default is already null

Reply:
thanks for verifying!